packages/runtime/container-runtime/src/summary/summarizerTypes.ts

```diff
@@ -213,6 +213,13 @@ export type SubmitSummaryResult =
 	| IUploadSummaryResult
 	| ISubmitSummaryOpResult;
 
+/** The stages of Summarize, used to describe how far progress succeeded in case of a failure at a later stage. */
+export type SummaryStage = SubmitSummaryResult["stage"] | "unknown";
```

Comment:
Does this expand out to be `"base" | "generate" | "upload" | "submit" | "unknown"`? I wonder if we should just make it somewhat more TypeSafe, or we can just have this be a string. Doesn't really matter to me. We should consolidate it either in this PR or a later PR.

Reply:
> Does this expand out to be "base" | "generate" | "upload" | "submit" | "unknown"?

Yes.  How would you make it more typesafe?  We shouldn't relax to merely string.

Reply:
+1 to what Mark said. The whole reason its written this way is to make it type safe.